technic/helpers.lua

```diff
@@ -64,19 +62,61 @@ function technic.swap_node(pos, name)
 	end
 end
 
+-- Wear down a tool depending on the remaining charge.
+local function set_RE_wear(stack, charge)
+	local stackdef = stack:get_definition()
+	if stackdef.max_charge and stackdef.wear_represents == "technic_RE_charge" then
+		if charge == 0 then
+			stack:set_wear(0)
+		else
+			stack:set_wear(math.max(1, math.min(65535, 65536 - math.floor(charge / stackdef.max_charge * 65535))))
+		end
+	end
+end
+
+function technic.set_RE_charge(stack, charge)
+	local meta = stack:get_meta()
+	charge = math.max(0, charge)
+	set_RE_wear(stack, charge)
+	meta:set_int("charge", charge)
+end
+
+function technic.get_RE_charge(stack)
+	local meta = stack:get_meta()
+	return meta:get_int("charge")
+end
 
---- Fully charge RE chargeable item.
--- Must be defined early to reference in item definitions.
-function technic.refill_RE_charge(stack)
-	local max_charge = technic.power_tools[stack:get_name()]
-	if not max_charge then return stack end
-	technic.set_RE_wear(stack, max_charge, max_charge)
-	local meta = minetest.deserialize(stack:get_metadata()) or {}
-	meta.charge = max_charge
-	stack:set_metadata(minetest.serialize(meta))
-	return stack
+function technic.use_RE_charge(stack, amount)
+	if technic.creative_mode then
+		-- Do not check charge in creative mode
+		return true
+	end
+	local meta = stack:get_meta()
+	local charge = meta:get_int("charge")
+	if charge < amount then
+		-- Not enough energy available
+		return false
+	end
+	charge = charge - amount
+	set_RE_wear(stack, charge)
+	meta:set_int("charge", charge)
+	-- Charge used successfully
+	return true
 end
 
+function technic.get_RE_item_load(load1, max_load)
+	if load1 == 0 then load1 = 65535 end
+	local temp = 65536 - load1
+	temp = temp / 65535 * max_load
+	return math.floor(temp + 0.5)
+end
+
+function technic.set_RE_item_load(load1, max_load)
+	if load1 == 0 then return 65535 end
+	local temp = load1 / max_load * 65535
+	temp = 65536 - temp
+	return math.floor(temp)
+end
```

Comment:
I think these functions can and should be removed, they're not used anywhere, and upstream API documentation actually states they are depreciated: https://github.com/minetest-mods/technic/blob/master/technic/doc/api.md#deprecated-functions

Reply:
Yeah, agreed if those are useless.
Did not look at these at all when I moved some functions from register.lua
But there was a comment, at least 8yo `-- Utility functions. Not sure exactly what they do.. water.lua uses the two first.`, 

I could either leave those out of this PR by moving them back to register.lua unmodified or just remove those, not exactly part of actual tool/api changes.
Kinda would like to leave irrelevant stuff out of this PR but maybe this is clear enough for removal... deprecation being documented long time ago

Reply:
Well they were for converting between tool wear and some value (charge), so I think they are related enough to be removed in this PR.

About that comment, here is the mentioned file, where the functions were used for charging tools: https://github.com/minetest-mods/technic/blob/ee5c6c23fab97f1bd1b217a654b72606938619b6/technic/water.lua

Reply:
Removed both 👍